src/graphql/candyMachine.ts

```diff
@@ -15,24 +15,223 @@ import fs from 'fs/promises';
-import { unzip } from '../lib/helpers/unZipFile.js';
-import { CACHE_PATH, EXTENSION_JSON } from '../../cli/helpers/constants.js';
+} from '../../cli/helpers/various';
+import { StorageType } from '../../cli/helpers/storage-type';
+import { download } from '../lib/helpers/downloadFile';
+import { unzip } from '../lib/helpers/unZipFile';
+import { CACHE_PATH, EXTENSION_JSON } from '../../cli/helpers/constants';
 import mkdirp from 'mkdirp';
 import { bs58 } from '@project-serum/anchor/dist/cjs/utils/bytes/index.js';
 import retry from 'async-retry';
-import exec from 'await-exec'
+import {
+  keypairIdentity,
+  Metaplex,
+  toMetaplexFile,
+  sol,
+} from '@metaplex-foundation/js';
+import { nftStorage } from '@metaplex-foundation/js-plugin-nft-storage';
+import {
+  Connection,
+  Keypair,
+  PublicKey,
+  PublicKeyInitData,
+} from '@solana/web3.js';
+import {
+  ammanMockStorage,
+  AmmanMockStorageDriver,
+} from '@metaplex-foundation/amman-client';
+import exec from 'await-exec';
 
 const dirname = path.resolve();
-const storageDir = "/app/tmp";
+
+const storageDir = process.env.APP_ENV === 'development' ? '/app/tmp' : 'tmp';
+
+// Downloads a zip file from zipUrl and returns the directory where zipUrl was unpacked
+const downloadZip = async (
+  zipUrl: string,
+  processId: string,
+): Promise<string> => {
+  // Unpack zip file
+  const processDir = `${storageDir}/${processId}`;
+  const zipFilesDir = `${storageDir}/${processId}/files`;
+  const zipFile = `${storageDir}/${processId}/files.zip`;
+  const dirExists = await fs
+    .stat(processDir)
+    .then(() => true)
+    .catch(() => false);
+
+  if (!dirExists) {
+    await mkdirp(`${storageDir}/${processId}`);
+    await download(zipUrl, zipFile);
+    await unzip(zipFile, zipFilesDir);
+  } else {
+    throw Error('zip dir already exists');
+  }
+
+  return zipFilesDir;
+};
+
+const SUPPORTED_MEDIA_FILE_EXTENSIONS = ['.png', '.jpg', '.jpeg', '.mp4'];
+
+const contentTypeForFileName = (
+  fileName: string,
+  supportedFileTypes = SUPPORTED_MEDIA_FILE_EXTENSIONS,
+) => {
+  const extensionToType = {
+    '.png': 'image/png',
+    '.jpg': 'image/jpeg',
+    '.jpeg': 'image/jpeg',
+    '.mp4': "video/mp4'",
+  };
+
+  const extension = fileName.split('.')[-1];
+  return extensionToType[extension];
+};
+
+const runUploadV2UsingHiddenSettings = async (
+  logger: winston.Logger,
+  processId: string,
+  args: {
+    collectionMint: string;
+    config: any;
+    callbackUrl: null | string;
+    guid?: string;
+    keyPair: string;
+    env: string;
+    filesZipUrl: string;
+    rpc: string;
+    setCollectionMint: boolean;
+  },
+) => {
+  const {
+    collectionMint: collectionMintParam,
+    setCollectionMint,
+    filesZipUrl,
+    config,
+    rpc,
+    env,
+    keyPair,
+  } = args;
+
+  logger.info('Uploading Candy Machine with hidden settings');
+
+  // setup connection and metaplex client
+  const bytes = bs58.decode(keyPair);
```

Comment:
I see two different base-58 packages being used across files in this PR. Would be good to be consistent everywhere.